lib/rules/no-invalid-position-at-import-rule/index.js

```diff
@@ -27,6 +40,10 @@ function rule(actual) {
 		root.walk((node) => {
 			const nodeName = node.name && node.name.toLowerCase();
 
+			if (node.type === 'atrule' && optionsMatches(options, 'ignoreAtRules', node.name)) {
```

Comment:
Shall we move this into the conditional below to keep everything together, i.e.:

```js
if (
	(node.type === 'atrule' &&
		nodeName !== 'charset' &&
		nodeName !== 'import' &&
		!optionsMatches(options, 'ignoreAtRules', node.name) &&
		isStandardSyntaxAtRule(node)) ||
	(node.type === 'rule' && isStandardSyntaxRule(node))
) {
	invalidPosition = true;

	return;
}
```

You're right to use `node.name` rather than `nodeName` as the `ignore*` options are case sensitive.

---

Incidentally, I wonder whether this rule can be refactored to avoid walking every node. Instead, walking just `import` at rules, and whether that'll [affect performance](https://stylelint.io/developer-guide/rules/#improve-the-performance-of-a-rule) or not. One for another time, though.

Reply:
True, maybe it's worth to refactor that rule. For now I've changed it as you suggest  - put every contition together